hack/generate-integration-compatibility-versions-test.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script generates a test case for the test/integration/apiserver/apiserver_test.go - TestFeatureGateCompatibilityEmulationVersion test.
+# Usage: `hack/update-integration-apiserver-test-feature-gate-compatibility-emulation-version.sh` 1.31.1
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Ensure the previous version is passed as an argument
+if [ "$#" -ne 1 ]; then
+  echo "Usage: $0 <PREV_VERSION> (eg: v1.31.1)"
+  exit 1
+fi
+
+PREV_VERSION="$1"
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::setup_env
+
+# Check if cluster with name $PREV_VERSION already exists
+if kind get clusters | grep -q "^${PREV_VERSION}$"; then
+  echo "Cluster ${PREV_VERSION} already exists, skipping creation."
+else
+  # Create a kind cluster for the previous versions if a cluster doesn't already exist
+  echo "Creating kind cluster ${PREV_VERSION}..."
+  kind create cluster --name "${PREV_VERSION}" --image kindest/node:"${PREV_VERSION}"
```

Comment:
Technically, relying on kind tags like this is unsafe [^1]:
1. Recommended to pin by digest for no surprises, compatibility, security
2. We don't necessarily pre-publish convenience builds for every patch release release anyhow

If the input is the previous version, we should probably locally build and run an image for that version (`kind build node image $version --image=kindest/node:$version`), it's a little slower but guaranteed to be compatible.

If the input is meant to control a kind image, we should just make $1 the image to run.

[^1]: https://kind.sigs.k8s.io/docs/user/quick-start/#creating-a-cluster:~:text=This%20will%20bootstrap,a%20kind%20release.

Reply:
Ah makes sense, thanks.  I've updated the script to accept the image to run as an input - $2